source/extensions/common/wasm/wasm.cc

```diff
@@ -852,6 +853,8 @@ const ProtobufWkt::Struct* Context::getMetadataStructProto(MetadataType type,
     return getRouteMetadataStructProto(decoder_callbacks_);
   case MetadataType::ResponseRoute:
     return getRouteMetadataStructProto(encoder_callbacks_);
+  case MetadataType::Node:
+    return &wasm_->local_info_.node().metadata();
```

Comment:
This is immutable for the duration of the proxy.
Looks like every call will do a round trip across the sandbox.
We should either cache it in the c++ library sandbox layer,
Or pass the immutable state into the sandbox in onStart() call.

Reply:
Made the node metadata cached by default in the API.

Added a test for node metadata.

PTAL